compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -347,6 +347,31 @@ pub fn eval_to_allocation_raw_provider<'tcx>(
         trace!("const eval: {:?} ({})", key, instance);
     }
 
+    if let ty::InstanceKind::Item(def_id) = key.value.instance.def
+        && matches!(tcx.def_kind(def_id), DefKind::Const | DefKind::AssocConst)
+    {
+        let ct = tcx.const_of_item(def_id).instantiate(tcx, key.value.instance.args);
```

Comment:
This is probably wrong because it assumes there's always a rhs of a const item to "get" :thinking: